agent/background.go

```diff
@@ -50,58 +50,46 @@ func (a *Agent) startHeartbeat(ctx context.Context, tc *taskContext, heartbeat c
 	}
 }
 
-func (a *Agent) startIdleTimeoutWatch(ctx context.Context, tc *taskContext, timeout chan struct{}, resetIdleTimeout chan time.Duration) {
+func (a *Agent) startIdleTimeoutWatch(ctx context.Context, tc *taskContext, cancel context.CancelFunc) {
 	timeoutInterval := defaultIdleTimeout
 	if a.opts.IdleTimeoutInterval != 0 {
 		timeoutInterval = a.opts.IdleTimeoutInterval
 	}
-	timer := time.NewTimer(timeoutInterval)
 
+	timer := time.NewTimer(timeoutInterval)
 	defer timer.Stop()
 	for {
 		select {
+		case <-ctx.Done():
+			grip.Info("Idle timeout watch canceled context")
+			return
 		case <-timer.C:
+			if taskTimeout := tc.getCurrentTimeout(); taskTimeout != 0 {
```

Comment:
`Agent.startIdleTimeoutWatch` sets its timeout interval by default to `defaultIdleTimeout` (20 minutes), but `taskContext.getCurrentTimeout` sets its timeout interval by default to `defaultCmdTimeout (2 hours). The docs in global.go define these this way:

> defaultCmdTimeout specifies the duration after which agent sends
> an IdleTimeout signal if a task's command does not run to completion.

> defaultIdleTimeout specifies the duration after which agent sends an
> IdleTimeout signal if a task produces no logs.

We allow users to override the defaultCmdTimeout by setting "timeout_secs" in their yaml file.

This means that we aren't actually using the idle timeout, just the command timeout. I think therefore we can delete the idle timeout completely.